### ISSUE TITLE
Blocking waiting for file lock on package cache

### ISSUE THREAD

**Issue by shijunti19** [rust-lang/cargo#9742]
**Problem**
now 2021-7-29
Blocking waiting for file lock on package cache
![image](https://user-images.githubusercontent.com/5392211/127426633-975dd4d5-ae53-4397-addf-13c02ab953ea.png)
![image](https://user-images.githubusercontent.com/5392211/127426645-294e35b4-55a7-44b6-b344-0a11fd1da16b.png)
It has been running like this for 3 hours

The system should write the. Package cache time record at regular intervals. If the interval is too long, it should interrupt and restart instead of deleting it artificially

**Steps**

1.cargo run
2.
3.

**Possible Solution(s)**

**Notes**

Output of `cargo version`:
cargo 1.54.0-nightly (070e459c2 2021-05-11)

**Comment by alexcrichton** [comment]
Thanks for the report! The file locking mechanism used by Cargo should automatically work if a previous Cargo process crashes or something like that, so if Cargo blocks on the file lock it *should* be because there's an active Cargo process somewhere else which is actually holding the lock. Historically though we've seen bugs in various filesystems and things like that which dilutes this guarantee.

It looks like you're on Windows, but do you know if there's anything nonstandard about the filesystem where  the `.package-cache` file is located on? Is it a standard Windows filesystem or perhaps formatted differently? We've also seen issues in the past where it's a network-mount or something like a VM shared folder or things like that.

**Comment by shijunti19** [comment]
@alexcrichton windows10  Content empty
CMD running the command often closes the window directly   Do not press Ctrl+ c
![image](https://user-images.githubusercontent.com/5392211/127516456-c0189d04-df6d-4824-af0e-50a0cc73aec2.png)

This is often encountered on windows because we can close the window directly

**Comment by alexcrichton** [comment]
Oh sorry but I can only read English so I'm not sure what to learn from the screenshot that you posted.

Could you describe the `D:\` drive, though? Is that an NTFS filesystem? Or some other Windows filesystem?

Also, can you test to see whether there are any `cargo.exe` processes running on your system that you're not aware of? (I typically have to open Task Manager to check that)

**Comment by shijunti19** [comment]
@alexcrichton 
 NTFS filesystem
I use vscode
This is the first time I run this morning.
He prompted this, and then I deleted the.package-cache.
It automatically works normally
![image](https://user-images.githubusercontent.com/5392211/127582224-fffd657f-275d-4d03-b813-d4134122407a.png)

![image](https://user-images.githubusercontent.com/5392211/127582738-ebca068f-6671-4435-82f9-a59550e6c83e.png)

**Comment by alexcrichton** [comment]
If you're using rust-analyzer a common occurrence is that rust-analyzer runs `cargo check` in the background and then if you run `cargo run` yourself those two commands will conflict (leading to frobbing on locks like you're seeing). That shouldn't cause indefinite blocking, though, like the 3+ hour wait time you initially reported.

With filesystems ruled out do you perhaps have any antivirus installed which you can temporarily disable to see if it fixes the issue?

**Comment by shijunti19** [comment]
@alexcrichton 
![image](https://user-images.githubusercontent.com/5392211/127721784-6476b6ac-324c-40ec-94da-1bcf3ac48078.png)
![image](https://user-images.githubusercontent.com/5392211/127721794-73edd043-6ed9-45ef-b5a0-d555592a9527.png)
It's this extension that causes it
I can't stop him yet. I have to wait until he finishes indexing
This extension should be optimized

**Comment by pmespresso** [comment]
@alexcrichton  So do you recommend just letting rust-analyzer run through? It seems almost every time I start a new project I run into this issue and rust-analyzer doesn't finish running particularly fast either...

**Comment by thadguidry** [comment]
I ran into this issue when crates.io was not accessible from my network.  I noticed that `.package-cache` was not readable by Windows programs or VSCode.  It definitely was somehow corrupted.  Before deleting the file and having Cargo make a new one, I tried with elevated permissions (CMD prompt run as Administrator) and also set the security for full access by any user on my Windows system.  Both did not help to allow Cargo to read the file (which did seem corrupted).  After deleting the `.package-cache` file from my E:\CARGO_HOME folder, I tried `cargo check` again within my Bevy project and it began to work:

```shell
E:\GitHubRepos\bevy>cargo -v check
    Updating crates.io index
    Fetch [======>                  ]  28.34%, 17.46MiB/s
```
